spectator/publisher_test.cc

```diff
@@ -83,6 +83,37 @@ TEST(Publisher, UnixBuffer) {
   unlink(path.c_str());
 }
 
+TEST(Publisher, UnixBufferTimeFlush) {
+  auto logger = spectator::DefaultLogger();
+  const auto* dir = first_not_null(std::getenv("TMPDIR"), "/tmp");
+  auto path = fmt::format("{}/testserver.{}", dir, getpid());
+  TestUnixServer server{path};
+  server.Start();
+  logger->info("Unix Server started on path {}", path);
+
+  // Set buffer size to a large value so that flushing is based on time
+  SpectatordPublisher publisher{fmt::format("unix:{}", path), 10000, std::chrono::seconds(5)};
```

Comment:
Not a blocker here, but ideally we'd have some sort of test time system instead of a real clock.